internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -729,6 +736,19 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 		update = true
 	}
 
+	if clusterInstall.Spec.DiskEncryption != nil {
+		params.DiskEncryption = &models.DiskEncryption{}
+		if cluster.DiskEncryption == nil { // true when current cluster configuration does not include disk encryption
+			cluster.DiskEncryption = &models.DiskEncryption{}
```

Comment:
Should we initialize DiskEncryption in this scenario? I mean, we can just invoke the upadateString statements only if cluster.DiskEncryption != nil, or do we prefer it initialized for later on?

Reply:
Yes, we should.
if `clusterInstall.Spec.DiskEncryption` is not `nil`, but `cluster.DiskEncryption` is `nil`, meaning that we see a new DiskEncryption config for a cluster that has no such configuration at the moment - we need to initialize struct so we can compare and change its attributes few lines below.

One thing that I need to check if, [per this comment](https://github.com/openshift/assisted-service/pull/2890#discussion_r744498898), is whether or not this scenario can still happen with a cluster with no DiskEncryption config set.

Reply:
but this section is nested within `clusterInstall.Spec.DiskEncryption != nil` , meaning I do have DiskEncryption in the spec. so I'd like to compare the configuration with the cluster config anyhow.
If I start with `if cluster.DiskEncryption != nil` , I'll basically repeat things..

example:
```go
	if clusterInstall.Spec.DiskEncryption != nil {
		if cluster.DiskEncryption != nil {
			updateString(swag.StringValue(clusterInstall.Spec.DiskEncryption.EnableOn), swag.StringValue(cluster.DiskEncryption.EnableOn), &params.DiskEncryption.EnableOn)
			updateString(swag.StringValue(clusterInstall.Spec.DiskEncryption.Mode), swag.StringValue(cluster.DiskEncryption.Mode), &params.DiskEncryption.Mode)
			if clusterInstall.Spec.DiskEncryption.TangServers != cluster.DiskEncryption.TangServers {
				params.DiskEncryption.TangServers = clusterInstall.Spec.DiskEncryption.TangServers
				update = true	
		} else {
				params.DiskEncryption.EnableOn = clusterInstall.Spec.DiskEncryption.EnableOn
				params.DiskEncryption.Mode = clusterInstall.Spec.DiskEncryption.Mode
				params.DiskEncryption.TangServers = clusterInstall.Spec.DiskEncryption.TangServers
			}
		}
	}
```

Reply:
yeah nm, isn't worth it.